src/components/molecules/TableHeader/TableHeader.tsx

```diff
@@ -36,13 +44,16 @@ const TableHeader: React.FC<TableHeaderProps> = ({
 
   const experience = useSelector(experienceSelector);
   const { recentVenueUsers } = useRecentVenueUsers();
+  const { isShown, show, hide } = useShowHide();
 
   const allTables = experience?.tables;
 
   const tableOfUser = seatedAtTable
     ? tables.find((table) => table.reference === seatedAtTable)
     : undefined;
 
+  const [error, setError] = useState<string>("");
```

Comment:
@denisdimitrov Have you seen `react-use`'s (which we already have in our deps) `useAsync` hook? Much cleaner pattern for handling async functions + state + error state in one hit.

- https://github.com/streamich/react-use
  - https://github.com/streamich/react-use/blob/master/docs/useAsync.md
  - https://github.com/streamich/react-use/blob/master/docs/useAsyncFn.md
  - https://github.com/streamich/react-use/blob/master/docs/useAsyncRetry.md

Reply:
This looks cool! I'll give it a try. I assume whenever there is an http error `state.error` will exists?

Reply:
Added: https://github.com/sparkletown/sparkle/pull/1377/commits/de659957079dd74136f26d29e9b952be12add510

Reply:
> I assume whenever there is an http error state.error will exists?

That's my understanding of it (I've used it but haven't actually tested that. Just trusted it will do what it's docs imply)

Reply:
Used in: https://github.com/sparkletown/sparkle/pull/1377/commits/4b7532e6f02068126c212a6f0dab1714aaaad55a